conan/tools/cmake/toolchain/blocks.py

```diff
@@ -549,19 +496,90 @@ def context(self):
-        endif()
-        if(DEFINED CONAN_EXE_LINKER_FLAGS)
-          string(APPEND CMAKE_EXE_LINKER_FLAGS_INIT " ${CONAN_EXE_LINKER_FLAGS}")
-        endif()
-        """)
+        {% if cxxflags %}
+        string(APPEND CMAKE_CXX_FLAGS_INIT " {{ cxxflags }}")
+        {% endif %}
+        {% if cflags %}
+        string(APPEND CMAKE_C_FLAGS_INIT " {{ cflags }}")
+        {% endif %}
+        {% if sharedlinkflags %}
+        string(APPEND CMAKE_SHARED_LINKER_FLAGS_INIT " {{ sharedlinkflags }}")
+        {% endif %}
+        {% if exelinkflags %}
+        string(APPEND CMAKE_EXE_LINKER_FLAGS_INIT " {{ exelinkflags }}")
+        {% endif %}
+    """)
+
+    def __init__(self, conanfile, toolchain):
+        super(CMakeFlagsInitBlock, self).__init__(conanfile, toolchain)
+        # Load all the predefined Conan C, CXX, etc. flags for CMakeToolchain
+        self._conan_conf = Conf()
+        self._process_conan_flags()
+
+    def _get_parallel_jobs_flags(self):
+        compiler = self._conanfile.settings.get_safe("compiler")
+        if compiler not in ("Visual Studio", "msvc") or "Visual" not in self._toolchain.generator:
+            return ""
+        jobs = build_jobs(self._conanfile)
+        if jobs:
+            return "/MP{}".format(jobs)
+
+    def _get_arch_flag(self):
+        arch_flag = architecture_flag(self._conanfile.settings)
+        if not arch_flag:
+            return ""
+        return arch_flag
+
+    def _get_glib_flag(self):
+        libcxx = self._conanfile.settings.get_safe("compiler.libcxx")
+        if not libcxx:
+            return ""
+
+        compiler = self._conanfile.settings.get_safe("compiler")
+        glib_flag = ""
+        if compiler == "apple-clang":
+            # In apple-clang 2 only values atm are "libc++" and "libstdc++"
+            glib_flag = "-stdlib={}".format(libcxx)
+        elif compiler == "clang" or compiler == "intel-cc":
+            if libcxx == "libc++":
+                glib_flag = "-stdlib=libc++"
+            elif libcxx == "libstdc++" or libcxx == "libstdc++11":
+                glib_flag = "-stdlib=libstdc++"
+            # FIXME, something to do with the other values? Android c++_shared?
+        elif compiler == "sun-cc":
+            glib_flag = {"libCstd": "Cstd",
+                         "libstdcxx": "stdcxx4",
+                         "libstlport": "stlport4",
+                         "libstdc++": "stdcpp"}.get(libcxx)
+            if glib_flag:
+                glib_flag = "-library={}".format(glib_flag)
+        return glib_flag
+
+    def _process_conan_flags(self):
+        """Calculating all the flags predefined by Conan"""
+        mp_flag = self._get_parallel_jobs_flags()
+        arch_flag = self._get_arch_flag()
+        glib_flag = self._get_glib_flag()
+
+        # Defining all the flags
+        self._conan_conf.define("tools.build:cxxflags", [arch_flag, glib_flag, mp_flag])
```

Comment:
I am not sure the aggregation has to be via conf, maybe it has to be defined via CMake instead (in the toolchain).
So the toolchain could define in one block/section some flags, as defined by the block (architecture, parallel, etc)

Then, other block could define the profile/conf variables, depending on the action can be added or overwritten.
But I don't love the idea of moving all the logic in "glibc", "architecture", etc blocks to a single block.

Lets see what others think.